### DISCUSSION TITLE
Bluetooth Pi4 e switchbot - potenziare segnale

### DISCUSSION THREAD

**Discussion by reiner348** [indomus/forum#1711] [Hardware & Firmware]
Ciao a tutti! 
Ho comprato 3 switchbot meter, integrati in home assistant usando sbm2mqtt, che funzionano comunicando via bluetooth col mio server raspberry Pi4.
Il problema è la potenza del segnale... Gli switchbot mi sembrano buoni, se uso l'app del cellulare li vedo bene, però tramite Pi4 sono molto limitati come range. Ho letto un po' su internet e sembra che il segnale bluetooth del Pi sia decisamente scarsino... Come si può fare per potenziarlo? Ho provato a collegare un dongle USB bluetooth, cinesata, che avevo in casa, e ad abilitare solo lui disattivando il modulo interno al pi (con i comandi hciconfig da terminale di raspbian), però la situazione non è migliorata...

**Comment by Scialla** [comment]
Prova a vedere qui:
https://indomus.it/componenti/realizzare-un-bridge-gateway-bluetooth-wi-fi-autonomo-con-nodemcu-esp32-e-esphome/

**Comment by reiner348** [comment]
Interessante ma complesso.. gli switchbot non sono nella lista, dovrei risalire a un bel po' di info oltre a dover guardare ESPhome. Per ora ho usato solo tasmota come firmware sugli ESP :) 
Fattibile forse, ma richiederà parecchio tempo! Intanto grazie della dritta!
